d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -201,16 +210,17 @@ public void getClient(Request request, RequestContext requestContext, Callback<T
         }
         else
         {
-          _log.debug("service hint found, using generic client for target: {}", targetService);
-
-          TransportClient transportClient = _state.getClient(serviceName, targetService.getScheme());
+          URI target = override == null ? targetService : URI.create(override + service.getPath());
```

Comment:
Do we need to make sure override variable ends with '/' here, to make sure that servicepath is added as a proper suffix?

Reply:
Given that override takes precedence, and there are many applications use targetService for now, we probably want to log a warning message here if both present?

Reply:
@nizarm's question, unit test shows that we don't have to. I will add another test to test with a '/' at the end.
@ChaoLinkedIn, I'll address this comment in the comment below.

Reply:
thanks